saiph/projection_test.py

```diff
@@ -350,6 +350,55 @@ def test_get_variable_contributions_calls_correct_subfunction(
     projection.get_variable_contributions(model, quanti_df)
 
 
+def test_get_variable_contributions_sum_is_100_famd(mixed_df: pd.DataFrame) -> None:
+    model = fit(mixed_df)
+    contributions = projection.get_variable_contributions(model, mixed_df)
+    summed_contributions = contributions.sum(axis=0)
+    assert_series_equal(summed_contributions, pd.Series([100.0] * 3), check_index=False)
+
+
+def test_get_variable_contributions_sum_is_100_with_col_weights_1_famd(mixed_df: pd.DataFrame) -> None:
+    model = fit(mixed_df, col_weights=[1.0, 1.0])
+    contributions = projection.get_variable_contributions(model, mixed_df)
+    summed_contributions = contributions.sum(axis=0)
+    assert_series_equal(summed_contributions, pd.Series([100.0] * 3), check_index=False)
+
+
+def test_get_variable_contributions_sum_is_100_with_col_weights_random_famd(mixed_df: pd.DataFrame) -> None:
+    model = fit(mixed_df, col_weights=[3.0, 2.0])
+    contributions = projection.get_variable_contributions(model, mixed_df)
+    summed_contributions = contributions.sum(axis=0)
+    assert_series_equal(summed_contributions, pd.Series([100.0] * 3), check_index=False)
+
+
+def test_get_variable_contributions_sum_is_100_with_col_weights_random_famd2(iris_df: pd.DataFrame) -> None:
+    model = fit(iris_df, col_weights=[3.0, 2.0, 0.5, 1.0, 2.1])
+    contributions = projection.get_variable_contributions(model, iris_df)
+    summed_contributions = contributions.sum(axis=0)
+    assert_series_equal(summed_contributions, pd.Series([100.0] * 7), check_index=False)
+
+
+def test_get_variable_contributions_sum_is_100_mca(quali_df: pd.DataFrame) -> None:
```

Comment:
Same comment as for famd, I think this should go in `mca_test.py` and test `mca.get_variable_contributions`